src/bin/parser.js

```diff
@@ -264,89 +240,31 @@ export default class Harmonic {
 
-            // write post html file
-            await fs.writeFileAsync(postFilePath, postHTMLFile);
-            console.log(clc.info(`Successfully generated post ${postPath}`));
+            await fs.writeFileAsync(publicFilePath, contentHTMLFile);
+            console.log(clc.info(`Successfully generated ${fileType} ${filePath}`));
 
-            posts[lang] = posts[lang] || [];
-            posts[lang].push(metadata);
+            generatedFiles[lang] = generatedFiles[lang] || [];
+            generatedFiles[lang].push(metadata);
         })));
-        return Helper.sortPosts(posts);
-    }
-
-    async generatePages(files) {
-        const langs = Object.keys(files);
-        const config = this.config;
-        const pages = {};
-        const tokens = config.header_tokens || ['<!--', '-->'];
-        const metadataDefaults = {
-            layout: 'page'
-        };
-
-        await* [].concat(...langs.map((lang) => files[lang].map(async (file) => {
-            const md = new MkMeta(path.join(this.sitePath, pagespath, lang, file));
-            md.defineTokens(tokens[0], tokens[1]);
-
-            const metadata = Helper.normalizeMetaData(md.metadata(), metadataDefaults);
-            const template = this.getTemplate(metadata.layout);
-            const filename = getFileName(file);
-
-            const pagePath = permalinks({
-                replacements: [{
-                    pattern: ':title',
-                    replacement: filename
-                },
-                {
-                    pattern: ':language',
-                    replacement: lang
-                }],
-                structure: getStructure(config.i18n.default, lang, config.pages_permalink)
-            });
-
-            metadata.file = postspath + file;
-            metadata.filename = filename;
-            metadata.link = pagePath;
-            metadata.lang = lang;
-            metadata.default_lang = config.i18n.default !== lang; // FIXME https://github.com/JSRocksHQ/harmonic/issues/169
-
-            const pageHTMLFile = template
-                .render({
-                    page: {
-                        content: md.markdown(),
-                        metadata
-                    },
-                    config,
-                    lang
-                })
-                .replace(/<!--[\s\S]*?-->/g, '');
-
-            metadata.content = pageHTMLFile;
 
-            const pageDirPath = path.join(this.sitePath, 'public', pagePath);
-            const pageFilePath = path.join(pageDirPath, 'index.html');
-            await mkdirpAsync(pageDirPath);
-
-            // write page html file
-            await fs.writeFileAsync(pageFilePath, pageHTMLFile);
-            console.log(clc.info(`Successfully generated page ${pagePath}`));
-
-            pages[lang] = pages[lang] || [];
-            pages[lang].push(metadata);
-        })));
-        return pages;
+        return fileType === 'post' ? this.sortByDate(generatedFiles) : generatedFiles;
```

Comment:
Yes, your right @UltCombo.
I didn't realize it could happen(return pages in different order).
I think sortByName could be a good approach, even if the user changes the date we could return the pages in the same order(Which I think it the behavior expected).
Above that, we could give an option to manually config the right order (only for pages).